translators/DirectionalLight.cpp

```diff
@@ -0,0 +1,310 @@
+    {
+        m_enableColorTemperature = numAttr.create("enableColorTemperature", "ect", MFnNumericData::kBoolean);
+        status = dirLight.addExtensionAttribute(m_enableColorTemperature);
+    }
+
+    AL_MAYA_CHECK_ERROR(status, errorString);
+    return status;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+MStatus DirectionalLight::import(const UsdPrim& prim, MObject& parent, MObject& createObj)
+{
+    MStatus status = MS::kSuccess;
+
+    MFnDirectionalLight fnDirectionLight;
+    createObj = fnDirectionLight.create(parent, true, false, &status);
+    AL_MAYA_CHECK_ERROR2(status, MString("unable to create directional light "));
+
+    TranslatorContextPtr ctx = context();
+
+    if(ctx)
+    {
+      ctx->insertItem(prim, createObj);
+    }
+    status = updateMayaAttributes(createObj, prim);
+
+    return status;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+UsdPrim DirectionalLight::exportObject(UsdStageRefPtr stage, MDagPath dagPath, const SdfPath& usdPath,
+                     const ExporterParams& params)
+{
+    MStatus status = MS::kSuccess;
+    MFnDirectionalLight fnDireLight(dagPath, &status);
+    AL_MAYA_CHECK_ERROR2(status, "Export: Failed to attach function set to directional light dagPath.");
+
+    const MObject direLightObj = fnDireLight.object(&status);
+    AL_MAYA_CHECK_ERROR2(status, "Export: Failed to retrieve object.");
+
+    bool result = updateUsdPrim(stage, usdPath, direLightObj);
+
+    //If there is animation keyed on this light
+    AnimationTranslator* animTranslator = params.m_animTranslator;
+    UsdLuxDistantLight usdLight(stage->GetPrimAtPath(usdPath));
+    if(animTranslator)
+    {
+      animTranslator->addPlug(MPlug(direLightObj, m_lightAngle), usdLight.GetAngleAttr(), true);
+      animTranslator->addPlug(MPlug(direLightObj, m_color), usdLight.GetColorAttr(), true);
+      animTranslator->addPlug(MPlug(direLightObj, m_intensity), usdLight.GetIntensityAttr(), true);
+      animTranslator->addPlug(MPlug(direLightObj, m_exposure), usdLight.GetExposureAttr(), true);
+      animTranslator->addPlug(MPlug(direLightObj, m_diffuse), usdLight.GetDiffuseAttr(), true);
+      animTranslator->addPlug(MPlug(direLightObj, m_specular), usdLight.GetSpecularAttr(), true);
+      animTranslator->addPlug(MPlug(direLightObj, m_normalize), usdLight.GetNormalizeAttr(), true);
+      animTranslator->addPlug(MPlug(direLightObj, m_enableColorTemperature), usdLight.GetEnableColorTemperatureAttr(), true);
+    }
+
+    return stage->GetPrimAtPath(usdPath);
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+MStatus DirectionalLight::preTearDown(UsdPrim& prim)
+{
+    TF_DEBUG(ALUSDMAYA_TRANSLATORS).Msg("DirectionalLightTranslator::preTearDown prim=%s\n", prim.GetPath().GetText());\
```

Comment:
You probably don't need the backslash here :)
```suggestion
    TF_DEBUG(ALUSDMAYA_TRANSLATORS).Msg("DirectionalLightTranslator::preTearDown prim=%s\n", prim.GetPath().GetText());
```

Reply:
Opps, copy/paste typo.